### PR TITLE
Add properties function to Cypher

### PR DESCRIPTION
**Pull request by boggle**

### CODE REVIEW

File: community/cypher/cypher-compiler-3.0/src/main/scala/org/neo4j/cypher/internal/compiler/v3_0/commands/expressions/PropertiesFunction.scala [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+import org.neo4j.cypher.internal.compiler.v3_0._
+import org.neo4j.cypher.internal.compiler.v3_0.helpers.IsMap
+import org.neo4j.cypher.internal.compiler.v3_0.pipes.QueryState
+import org.neo4j.cypher.internal.compiler.v3_0.symbols.SymbolTable
+import org.neo4j.cypher.internal.frontend.v3_0.CypherTypeException
+import org.neo4j.cypher.internal.frontend.v3_0.symbols._
+
+case class PropertiesFunction(a: Expression) extends NullInNullOutExpression(a) {
+  def symbolTableDependencies: Set[String] = a.symbolTableDependencies
```
Comment (systay) [inline community/cypher/cypher-compiler-3.0/src/main/scala/org/neo4j/cypher/internal/compiler/v3_0/commands/expressions/PropertiesFunction.scala]: `override` is nice to add

**Comment by systay** [comment]
Love it. Took the liberty of adding a commit to your PR. Please slap me around a bit with a large trout if you disagree.

**Comment by jexp** [comment]
Love it too. Makes it much clearer what you want to express.

File: community/cypher/acceptance/src/test/scala/org/neo4j/internal/cypher/acceptance/FunctionsAcceptanceTest.scala [hunk 2]
```diff
@@ -208,6 +208,48 @@ class FunctionsAcceptanceTest extends ExecutionEngineFunSuite with NewPlannerTes
...
+    // When
+    val result = executeScalarWithAllPlanners[Map[String,Any]](
+      "RETURN properties({name: 'Popeye', level: 9001})"
+    )
+
+    // Then
+    result should equal(Map("name" -> "Popeye", "level" -> 9001))
+  }
+
+  test("properties(null) should be null") {
```
Comment (Mats-SX) [inline community/cypher/acceptance/src/test/scala/org/neo4j/internal/cypher/acceptance/FunctionsAcceptanceTest.scala]: I'd like to see some more tests like these: what kind of error is expected for `properties("")`, or other types. Perhaps in `SemanticErrorAcceptanceTest`.

Comment (boggle) [inline community/cypher/acceptance/src/test/scala/org/neo4j/internal/cypher/acceptance/FunctionsAcceptanceTest.scala]: What would be sufficient, a test for every type (ignoring nested list types)?  We don't do that for other functions, in this case I felt testing lists and scalars and maps is alright but perhaps we should do more?

Comment (Mats-SX) [inline community/cypher/acceptance/src/test/scala/org/neo4j/internal/cypher/acceptance/FunctionsAcceptanceTest.scala]: I think testing a few\* other types is sufficient, just to make sure that there is a well-defined behaviour for incompatible types. You have a few tests down below, but none on the acceptance level (query in -> error out).

*string, list, integer, for example.

**Pull request merged**